desc/compute/_neoclassical.py

```diff
@@ -68,85 +86,46 @@ def _compute(fun, fun_data, data, grid, num_pitch, simp=False, reduce=True):
         Data to provide to ``fun``. This dict will be modified.
     data : dict[str, jnp.ndarray]
         DESC data dict.
+    theta : jnp.ndarray
+        Shape (num rho, X, Y).
+        DESC coordinates θ sourced from the Clebsch coordinates
+        ``FourierChebyshevSeries.nodes(X,Y,rho,domain=(0,2*jnp.pi))``.
     simp : bool
         Whether to use an open Simpson rule instead of uniform weights.
-    reduce : bool
-        Whether to compute mean over α and expand to grid.
-        Default is true.
+    surf_batch_size : int
+        Number of flux surfaces with which to compute simultaneously.
+        Default is ``1``.
+
     """
-    pitch_inv, pitch_inv_weight = Bounce1D.get_pitch_inv_quad(
+    for name in Bounce2D.required_names:
+        fun_data[name] = data[name]
+    fun_data.pop("iota", None)
+    for name in fun_data:
+        fun_data[name] = Bounce2D.fourier(Bounce2D.reshape(grid, fun_data[name]))
+    fun_data["iota"] = grid.compress(data["iota"])
+    fun_data["theta"] = theta
+    fun_data["pitch_inv"], fun_data["pitch_inv weight"] = Bounce2D.get_pitch_inv_quad(
         grid.compress(data["min_tz |B|"]),
         grid.compress(data["max_tz |B|"]),
         num_pitch,
         simp=simp,
     )
-
-    def foreach_rho(x):
-        # using same λ values for every field line α on flux surface ρ
-        x["pitch_inv"] = pitch_inv
-        x["pitch_inv weight"] = pitch_inv_weight
-        return imap(fun, x)
-
-    for name in Bounce1D.required_names:
-        fun_data[name] = data[name]
-    for name in fun_data:
-        fun_data[name] = Bounce1D.reshape_data(grid, fun_data[name])
-    out = imap(foreach_rho, fun_data)
-    return grid.expand(_alpha_mean(out)) if reduce else out
+    return grid.expand(batch_map(fun, fun_data, surf_batch_size))
 
 
-@register_compute_fun(
-    name="fieldline length",
-    label="\\int_{\\zeta_{\\mathrm{min}}}^{\\zeta_{\\mathrm{max}}}"
-    " \\frac{d\\zeta}{|B^{\\zeta}|}",
-    units="m / T",
-    units_long="Meter / tesla",
-    description="(Mean) proper length of field line(s)",
-    dim=1,
-    params=[],
-    transforms={"grid": []},
-    profiles=[],
-    coordinates="r",
-    data=["B^zeta"],
-    resolution_requirement="z",
-    source_grid_requirement={"coordinates": "raz", "is_meshgrid": True},
-)
-def _fieldline_length(data, transforms, profiles, **kwargs):
-    grid = transforms["grid"].source_grid
-    L_ra = simpson(
-        y=grid.meshgrid_reshape(1 / data["B^zeta"], "arz"),
-        x=grid.compress(grid.nodes[:, 2], surface_label="zeta"),
-        axis=-1,
+def _dH(data, B, pitch):
```

Comment:
I know it's called H in Nemov et al but can we add more letters/words to this. Otherwise, it may become a problem later.
Perhaps  _dH_ripple?
Not urgent.

Reply:
done in different PR